test/ngMock/angular-mocksSpec.js

```diff
@@ -1943,10 +1943,30 @@ describe('ngMock', function() {
         );
         they('should ignore query param when matching in ' + routeShortcut + ' $prop method', methods,
           function() {
-            hb[routeShortcut](this, '/route/:id').respond('path');
+            var paramsSpy = jasmine.createSpy('paramsSpy');
+            hb[routeShortcut](this, '/route/:id').respond(function(method, url, data, headers, params) {
+              paramsSpy(params);
+              // status, response, headers, statusText, xhrStatus
+              return [200, 'path', { 'x-header': 'foo' }, 'OK', 'complete'];
+            });
             hb(this, '/route/123?q=str&foo=bar', undefined, callback);
             hb.flush();
-            expect(callback).toHaveBeenCalledOnceWith(200, 'path', '', '', 'complete');
+            expect(callback).toHaveBeenCalledOnceWith(200, 'path', 'x-header: foo', 'OK', 'complete');
+            expect(paramsSpy).toHaveBeenCalledOnceWith({id: '123', q: 'str', foo: 'bar'});
+          }
+        );
+        they('should ignore query param when matching eager parameters in ' + routeShortcut + ' $prop method', methods,
+          function() {
+            var paramsSpy = jasmine.createSpy('paramsSpy');
+            hb[routeShortcut](this, '/route/:id*').respond(function(method, url, data, headers, params) {
```

Comment:
How about adding a test without the `*` (for good measure)?

Reply:
I misread this question before. There already is a test without `*`. I added a test with `*`. I thought you were asking about a test without any routing parameters at all.

Reply:
🙈